pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -503,6 +503,14 @@ void connectionOpened(final ClientCnx cnx) {
                 cnx.channel().close();
                 return null;
             }
+            
+            if (client.getConfiguration().getConnectionsPerBroker() == 0) {
```

Comment:
How this would fix cases in which a producer is created and then closed? Or the connection used to do a topic lookup?